File: examples/python/src/06_auth_get_user.py
import json
import os
from client import McpSseClient, env

def main():
    email = env("DEMO_EMAIL", "user@example.com")
    code = os.getenv("DEMO_CODE")
    auth_key = os.getenv("DEMO_AUTH_KEY")

    c = McpSseClient(env("MCP_BASE_URL"))

    if not code and not auth_key:
        print("Step 1: request one-time code (valid ~10 minutes)")
        step1 = c.call_tool("getAuthKey", {"email": email})
        print(json.dumps(step1, indent=2))
        print("\nNow set DEMO_CODE in .env and rerun this script.")
        return

    if not auth_key and code:
        print("Step 2: exchange code for authKey")
        step2 = c.call_tool("getAuthKey", {"email": email, "code": code})
        print(json.dumps(step2, indent=2))
        print("\nNow set DEMO_AUTH_KEY from the response and rerun to call getUser.")
        return

    print("Fetching user data with authKey")
    user_res = c.call_tool("getUser", {"email": email, "authKey": auth_key})
    print(json.dumps(user_res, indent=2))

if __name__ == "__main__":
    main()

File: examples/php/06_auth_get_user.php
<?php
declare(strict_types=1);

require __DIR__ . '/client.php';

$email = env('DEMO_EMAIL', 'user@example.com');
$code = getenv('DEMO_CODE') ?: null;
$authKey = getenv('DEMO_AUTH_KEY') ?: null;

if ($code === null && $authKey === null) {
    echo "Step 1: request one-time code (valid ~10 minutes)" . PHP_EOL;
    $step1 = callTool('getAuthKey', ['email' => $email]);
    echo json_encode($step1, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES) . PHP_EOL;
    echo PHP_EOL . "Now set DEMO_CODE in .env and rerun this script." . PHP_EOL;
    exit(0);
}

if ($authKey === null && $code !== null) {
    echo "Step 2: exchange code for authKey" . PHP_EOL;
    $step2 = callTool('getAuthKey', ['email' => $email, 'code' => $code]);
    echo json_encode($step2, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES) . PHP_EOL;
    echo PHP_EOL . "Now set DEMO_AUTH_KEY from the response and rerun to call getUser." . PHP_EOL;
    exit(0);
}

echo "Fetching user data with authKey" . PHP_EOL;
$userRes = callTool('getUser', ['email' => $email, 'authKey' => $authKey]);

echo json_encode($userRes, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES) . PHP_EOL;
